h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -650,7 +651,7 @@ private ByteBuffer sendPartial() {
       long ns = System.nanoTime();
       while( _bb.hasRemaining() ) {
         _chan.write(_bb);
-        if( RANDOM_TCP_DROP != null &&_chan instanceof SocketChannel && RANDOM_TCP_DROP.nextInt(100) == 0 )
+        if( RANDOM_TCP_DROP != null && SocketChannelUtils.socketChannel(_chan) && RANDOM_TCP_DROP.nextInt(100) == 0 )
```

Comment:
I would propose to rename `socketChannel` to `isSocketChannel` - just for readability.